src/Bicep.Core/Registry/Oci/OciArtifactReference.cs

```diff
@@ -57,22 +59,43 @@ private OciArtifactReference(string registry, string repository, string? tag, st
             ? $"{this.Registry}/{this.Repository}:{this.Tag}"
             : $"{this.Registry}/{this.Repository}@{this.Digest}";
 
-        public string FullyQualifiedReference => ArtifactId;
+        /// <summary>
+        /// Gets the type of artifact reference. Either module or provider.
+        /// </summary>
+        public ArtifactType Type { get; }
+
+        public override string UnqualifiedReference => ArtifactId;
```

Comment:
Looks like you're renaming `FullyQualifiedReference` -> `UnqualifiedReference` - the two names seem to directly conflict with each other. Is this correct?

Reply:
`ArtifactReference.cs` defines the `FullyQualifiedReference ` and `UnqualifiedReference` as following:
https://github.com/Azure/bicep/blob/7b6c1ff20f1242e3e71c6e463943356f6ad0209c/src/Bicep.Core/Registry/ArtifactReference.cs#L26-L34

In this case, I am setting the value of `UnqualifiedReference` to be the `ArtifactId` which I believe is the desired value, can you explain why you think is not the appropriate value for this field?

Reply:
> can you explain why you think is not the appropriate value for this field?

I wasn't making a statement on whether the old or new behavior was correct - was just noticing that there was a change in behavior. That led me to wonder:
1. Was this change intentional?
2. If so, does that imply the original behavior was incorrect?
3. If the original behavior was incorrect, are there places in the code which rely on this incorrectness?

Reply:
I think the naming here is a bit conflating, so I addressed this issue by using a new terminology. Resolving but feel free to open in case this is not an acceptable resoution